lib/new_relic/agent/transaction.rb

```diff
@@ -260,20 +257,26 @@ def initialize(category, options)
         end
       end
 
-      def parent_thread_id
-        ::Thread.current.nr_parent_thread_id if ::Thread.current.respond_to?(:nr_parent_thread_id)
+      def state
```

Comment:
This just makes sense the transaction is always using the correct state object. we were storing the state on the transaction originally, which wasn't causing any issues, but this is just a preemptive safety change for just in case situations.